app/uk/gov/hmrc/fileupload/write/infrastructure/EventStore.scala

```diff
@@ -80,19 +86,25 @@ class MongoEventStore(mongo: () => DB with DBMetaCommands, metrics: MetricRegist
       case e =>
         Xor.left(NotSavedError(e.getMessage))
     }.map { e =>
-      timer.stop()
+      context.stop()
       e
     }
   }
 
 
   override def unitsOfWorkForAggregate(streamId: StreamId): Future[GetResult] = {
-    val timer = metrics.timer("MongoEventStoreRead").time()
+    val context = readTimer.time()
     collection.find(BSONDocument("streamId" -> streamId.value)).cursor[UnitOfWork]().collect[List]().map { l =>
       val sortByVersion = l.sortBy(_.version.value)
       Xor.right(sortByVersion)
     }.recover { case e => Xor.left(GetError(e.getMessage)) }.map { e =>
-      timer.stop()
+      val elapsedNanos = context.stop()
+      val elapsed = FiniteDuration(elapsedNanos, TimeUnit.NANOSECONDS)
+
+      if (elapsed.>(FiniteDuration(20, TimeUnit.SECONDS))) {
+        Logger.warn(s"unitsOfWorkForAggregate query on stream $streamId took $elapsed")
```

Comment:
let's just put collection name "events" here so easier to repeat

Logger.warn(s"unitsOfWorkForAggregate: events.find by streamId=$streamId took $elapsed sec")

~10 sec would be already too high (some clients were timeouting with 20 secs)